From a2e1f8ac63b8f7afc2397272baf93a3779fef291 Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:21:19 -0400
Subject: [net] netfilter: masquerading needs to be independent of x_tables in Kconfig

Message-id: <20141021162119.22888.68974.stgit@localhost.localdomain>
Patchwork-id: 98017
O-Subject: [RHEL7 net PATCH 120/124] netfilter: masquerading needs to be independent of x_tables in Kconfig
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: 0bbe80e571c7b866afd92a98edd32a969467a7a9

 commit 0bbe80e571c7b866afd92a98edd32a969467a7a9
 Author: Pablo Neira Ayuso <pablo@netfilter.org>
 Date:   Thu Sep 11 17:51:27 2014 +0200

    netfilter: masquerading needs to be independent of x_tables in Kconfig

    Users are starting to test nf_tables with no x_tables support. Therefore,
    masquerading needs to be indenpendent of it from Kconfig.

    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv4/netfilter/Kconfig b/net/ipv4/netfilter/Kconfig
index b4e2e60..fb82242 100644
--- a/net/ipv4/netfilter/Kconfig
+++ b/net/ipv4/netfilter/Kconfig
@@ -88,6 +88,21 @@ config NFT_CHAIN_NAT_IPV4
    packet transformations such as the source, destination address and
    source and destination ports.
 
+config NF_NAT_MASQUERADE_IPV4
+ tristate "IPv4 masquerade support"
+ help
+   This is the kernel functionality to provide NAT in the masquerade
+   flavour (automatic source address selection).
+
+config NFT_MASQ_IPV4
+ tristate "IPv4 masquerading support for nf_tables"
+ depends on NF_TABLES_IPV4
+ depends on NFT_MASQ
+ select NF_NAT_MASQUERADE_IPV4
+ help
+   This is the expression that provides IPv4 masquerading support for
+   nf_tables.
+
 config NF_NAT_SNMP_BASIC
  tristate "Basic SNMP-ALG support"
  depends on NF_CONNTRACK_SNMP
@@ -245,18 +260,6 @@ config IP_NF_NAT
 
 if IP_NF_NAT
 
-config NF_NAT_MASQUERADE_IPV4
- tristate "IPv4 masquerade support"
- help
- This is the kernel functionality to provide NAT in the masquerade
- flavour (automatic source address selection).
-
-config NFT_MASQ_IPV4
- tristate "IPv4 masquerading support for nf_tables"
- depends on NF_TABLES_IPV4
- depends on NFT_MASQ
- select NF_NAT_MASQUERADE_IPV4
-
 config IP_NF_TARGET_MASQUERADE
  tristate "MASQUERADE target support"
  select NF_NAT_MASQUERADE_IPV4
diff --git a/net/ipv6/netfilter/Kconfig b/net/ipv6/netfilter/Kconfig
index 8578ef8..1b53023 100644
--- a/net/ipv6/netfilter/Kconfig
+++ b/net/ipv6/netfilter/Kconfig
@@ -71,6 +71,21 @@ config NFT_CHAIN_NAT_IPV6
    packet transformations such as the source, destination address and
    source and destination ports.
 
+config NF_NAT_MASQUERADE_IPV6
+ tristate "IPv6 masquerade support"
+ help
+   This is the kernel functionality to provide NAT in the masquerade
+   flavour (automatic source address selection) for IPv6.
+
+config NFT_MASQ_IPV6
+ tristate "IPv6 masquerade support for nf_tables"
+ depends on NF_TABLES_IPV6
+ depends on NFT_MASQ
+ select NF_NAT_MASQUERADE_IPV6
+ help
+   This is the expression that provides IPv4 masquerading support for
+   nf_tables.
+
 endif # NF_NAT_IPV6
 
 config IP6_NF_IPTABLES
@@ -261,18 +276,6 @@ config IP6_NF_NAT
 
 if IP6_NF_NAT
 
-config NF_NAT_MASQUERADE_IPV6
- tristate "IPv6 masquerade support"
- help
-  This is the kernel functionality to provide NAT in the masquerade
-  flavour (automatic source address selection) for IPv6.
-
-config NFT_MASQ_IPV6
- tristate "IPv6 masquerade support for nf_tables"
- depends on NF_TABLES_IPV6
- depends on NFT_MASQ
- select NF_NAT_MASQUERADE_IPV6
-
 config IP6_NF_TARGET_MASQUERADE
  tristate "MASQUERADE target support"
  select NF_NAT_MASQUERADE_IPV6
-- 
1.7.1